contracts/UbeswapOrderBook.sol

```diff
@@ -69,37 +62,17 @@ contract UbeswapOrderBook is OrderBook, Ownable {
         feeRecipient = _feeRecipient;
     }
 
-    /// @notice Admin function to change the subsidy rate for a makerToken
-    /// @param _token The makerToken
-    /// @param _subsidyRate The new subsidy rate
-    function changeSubsidyRate(address _token, uint256 _subsidyRate)
+    /// @notice Admin function to change the reward distributor contract
+    /// @param _rewardDistributor The new reward distributor. 0 address will disable rewards
+    function changeRewardDistributor(IOrderRewardDistributor _rewardDistributor)
         external
         onlyOwner
     {
-        // solhint-disable-next-line reason-string
-        require(
-            _subsidyRate <= PCT_DENOMINATOR,
-            "UOB: Subsidy exceeds PCT_DENOMINATOR"
+        emit RewardDistributorChanged(
+            address(rewardDistributor),
+            address(_rewardDistributor)
         );
-        emit SubsidyRateChanged(_token, subsidyRate[_token], _subsidyRate);
-        subsidyRate[_token] = _subsidyRate;
-    }
-
-    /// @notice Admin function to change the subsidy currency
-    /// @param _subsidyCurrency The new subsidy currency
-    function changeSubsidyCurrency(IERC20 _subsidyCurrency) external onlyOwner {
-        emit SubsidyCurrencyChanged(
-            address(subsidyCurrency),
-            address(_subsidyCurrency)
-        );
-        subsidyCurrency = _subsidyCurrency;
-    }
-
-    /// @notice Admin function to rescue any ERC20 tokens in the contract
-    /// @param _token The currency to rescue
-    function rescueERC20(IERC20 _token, uint256 _amount) external onlyOwner {
-        emit ERC20Rescued(address(_token), _amount);
-        _token.safeTransfer(msg.sender, _amount);
+        rewardDistributor = _rewardDistributor;
     }
 
     function broadcastOrder(
```

Comment:
the signature for `broadcastOrder` should be changed to:
```
function broadcastOrder(
  LimitOrderProtocol.Order _order,
  bytes calldata _signature,
  bytes calldata _orderInteraction)
```

`OrderInteraction` then get decoded to [address, OrderNotifier data]

The reward distributor is then notified of the order + the input order notifier data, and reward distributor can do w/e it wants once its notified by a whitelisted OrderBook of a new order